kpops/components/common/kubernetes_model.py

```diff
@@ -42,6 +50,122 @@ class ImagePullPolicy(str, enum.Enum):
+        match self.operator:
+            case NodeSelectorOperator.IN | NodeSelectorOperator.NOT_IN:
+                assert (
+                    self.values
+                ), "If the operator is In or NotIn, the values array must be non-empty."
+            case NodeSelectorOperator.EXISTS | NodeSelectorOperator.DOES_NOT_EXIST:
+                assert not self.values, "If the operator is Exists or DoesNotExist, the values array must be empty."
+            case NodeSelectorOperator.GT | NodeSelectorOperator.LT:
+                assert (
+                    len(self.values) == 1
+                ), "If the operator is Gt or Lt, the values array must have a single element, which will be interpreted as an integer."
+        return self
+
+
+class NodeSelectorTerm(DescConfigModel, CamelCaseConfigModel):
+    match_expressions: list[NodeSelectorRequirement] | None = None
+    match_fields: list[NodeSelectorRequirement] | None = None
+
+
+class NodeSelector(DescConfigModel, CamelCaseConfigModel):
+    node_selector_terms: list[NodeSelectorTerm]
+
+
+class PreferredSchedulingTerm(DescConfigModel, CamelCaseConfigModel):
+    preference: NodeSelectorTerm
+    weight: Weight
+
+
+class NodeAffinity(DescConfigModel, CamelCaseConfigModel):
+    required_during_scheduling_ignored_during_execution: NodeSelector | None = None
+    preferred_during_scheduling_ignored_during_execution: (
+        list[PreferredSchedulingTerm] | None
+    ) = None
+
+
+class LabelSelectorOperator(str, enum.Enum):
+    IN = "In"
+    NOT_IN = "NotIn"
+    EXISTS = "Exists"
+    DOES_NOT_EXIST = "DoesNotExist"
+
+
+class LabelSelectorRequirement(DescConfigModel, CamelCaseConfigModel):
+    key: str
+    operator: LabelSelectorOperator
+    values: list[str] = pydantic.Field(
+        default=[],
+        description="An array of string values. If the operator is In or NotIn, the values array must be non-empty. If the operator is Exists or DoesNotExist, the values array must be empty. This array is replaced during a strategic merge patch.",
+    )
+
+    @model_validator(mode="after")
+    def validate_values(self) -> Self:
+        match self.operator:
+            case LabelSelectorOperator.IN | LabelSelectorOperator.NOT_IN:
+                assert (
+                    self.values
+                ), "If the operator is In or NotIn, the values array must be non-empty."
+            case LabelSelectorOperator.EXISTS | LabelSelectorOperator.DOES_NOT_EXIST:
+                assert not self.values, "If the operator is Exists or DoesNotExist, the values array must be empty."
+        return self
+
+
+class LabelSelector(DescConfigModel, CamelCaseConfigModel):
+    match_labels: dict[str, str] | None = None
+    match_expressions: list[LabelSelectorRequirement] | None = None
+
+
+class PodAffinityTerm(DescConfigModel, CamelCaseConfigModel):
+    label_selector: LabelSelector | None = None
+    topology_key: str
+    namespaces: list[str] | None = None
+    weight: Weight
```

Comment:
Are you sure there is a weight here? Because it gives me a validation error with out antiaffinity we used for years. There is a weight one level up in WeightedPodAffinityTerm, maybe you confused sth here

Reply:
Yeah it does not exists here https://github.com/kubernetes/kubernetes/blob/a499facee693a1a83daadb82d88f7b51d324ffc5/api/openapi-spec/v3/api__v1_openapi.json#L5049